### ISSUE TITLE
[Feature request] Multipoint cursors

### ISSUE THREAD

**Issue by MaxGraey** [google/xi-editor#763]
According this [frontend docs](http://google.github.io/xi-editor/docs/frontend-protocol.html#gesture) xi-editor allowing several types of gestures like `range_select` __(Shift+click)__, `multi_word_select` and etc, but unfortunately can't support multicursor (cursor shared between several lines but have same y-coordinate) like this:

<img width="399" alt="multicursor" src="https://user-images.githubusercontent.com/1301959/43211910-6086c84a-903b-11e8-8faa-60ca6fe15b6a.png">

In most editors this can be selected via (Alt/Option + vertical drag). I tried `multi_word_select`, `multi_line_select` and `line_select` but any of this types not works as I expected.

Do you plan something like that? May be something like new type `multi_point_select`?

**Comment by raphlinus** [comment]
I think this request might cover the alt-drag gesture to extend a cursor across multiple lines. That shouldn't be too hard to add, maybe a "help wanted" feature.

**Comment by MaxGraey** [comment]
@raphlinus It will be great. Thanks!

**Comment by cmyr** [comment]
@MaxGraey can you clarify the behaviour that you're describing? Is this using the _mouse_ to select multiple lines, with each line in the selection have its own cursor?

Currently you can add a cursor at the same position on the previous or next line with <kbd>option</kbd><kbd>shift</kbd><kbd>↑</kbd>, but you can't add multiple cursors by dragging. This is definitely doable, though.

**Comment by MaxGraey** [comment]
Yes, selecting with <kbd>Option</kbd> or <kbd>Alt</kbd> + mouse drag to up or down which create vertical cursors line.

**Comment by jansol** [comment]
Hmm actually this seems like mostly a client issue? Handling drag gestures and sending new cursor requests to core.

**Comment by ryanbloom** [comment]
I think of this feature as a special case of rectangle selection (where you can option+drag horizontally as well as vertically). Rectangle selection looks kind of tricky; it'll probably require some width-measurement stuff that I haven't figured out yet. But at some point I think we'll want it to be a type of gesture in the core.
